actor/builtin/miner/miner.go

```diff
@@ -950,8 +950,7 @@ func (ma *Actor) SubmitPoSt(ctx exec.VMContext, poStProofs []types.PoStProof, do
 		// Update miner power to the amount of data actually proved
 		// during the last proving period.
 		oldPower := state.Power
-		// TODO subtract total faulted size from ProvingSet size #2889
-		newPower := types.NewBytesAmount(uint64(state.ProvingSet.Size())).Mul(state.SectorSize)
+		newPower := types.NewBytesAmount(uint64(state.ProvingSet.Size() - faults.SectorIds.Size())).Mul(state.SectorSize)
```

Comment:
Blocking: faults need to be subtracted from the sector commitments. Otherwise power will be restored on the next submission. See
``` go
	// update sector sets and proving set
	miner.Sectors.Subtract(done)
	miner.Sectors.Subtract(permLostSet)
```
in [the spec](https://github.com/filecoin-project/specs/blob/master/actors.md).